### PR TITLE
Chenyan98/2

### PR DESCRIPTION
**Pull request by ChenYan98**

### CODE REVIEW

File: TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
-using System.Linq;
-using System.Text;
 using System.Threading.Tasks;
 using Microsoft.Owin.Security.Jwt;
-using Microsoft.Owin.Security;
-using System.Net.Http;
 using System.IdentityModel.Tokens;
 using Microsoft.IdentityModel.Protocols;
-using System.Threading;
+using System.Net.Http;
```
Comment (polita) [inline TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs]: Please alphabetize the Using statements.

File: TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs [hunk 2]
```diff
@@ -64,32 +46,15 @@ public IEnumerable<SecurityToken> SecurityTokens
...
-                _synclock.EnterReadLock();
-                try
-                {
-                    return _tokens;
-                }
-                finally
-                {
-                    _synclock.ExitReadLock();
-                }
+                return RetrieveMetadata().Result.SigningTokens;
```
Comment (polita) [inline TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs]: We're still calling .Result on an async method. Is that the best way, @brentschmaltz?

File: TaskService/TaskService.csproj [hunk 3]
```diff
@@ -44,9 +44,9 @@
   </PropertyGroup>
   <ItemGroup>
     <Reference Include="Microsoft.CSharp" />
-    <Reference Include="Microsoft.IdentityModel.Protocol.Extensions, Version=1.0.2.28, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\Microsoft.IdentityModel.Protocol.Extensions.1.0.2.202250711\lib\net45\Microsoft.IdentityModel.Protocol.Extensions.dll</HintPath>
+    <Reference Include="Microsoft.IdentityModel.Protocol.Extensions, Version=1.0.3.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.IdentityModel.Protocol.Extensions.1.0.3.308261200\lib\net45\Microsoft.IdentityModel.Protocol.Extensions.dll</HintPath>
+      <Private>True</Private>
```
Comment (polita) [inline TaskService/TaskService.csproj]: Please revert changes to this file.

File: TaskService/Web.config [hunk 4]
```diff
@@ -56,7 +56,7 @@
       </dependentAssembly>
       <dependentAssembly>
         <assemblyIdentity name="System.IdentityModel.Tokens.Jwt" publicKeyToken="31bf3856ad364e35" culture="neutral" />
-        <bindingRedirect oldVersion="0.0.0.0-4.0.2.28" newVersion="4.0.2.28" />
+        <bindingRedirect oldVersion="0.0.0.0-4.0.30826.1200" newVersion="4.0.30826.1200" />
```
Comment (polita) [inline TaskService/Web.config]: Why are we creating a binding redirect here?

File: TaskService/Web.config [hunk 5]
```diff
@@ -70,6 +70,10 @@
         <assemblyIdentity name="Microsoft.Owin" publicKeyToken="31bf3856ad364e35" culture="neutral" />
         <bindingRedirect oldVersion="0.0.0.0-3.0.1.0" newVersion="3.0.1.0" />
       </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="Microsoft.IdentityModel.Protocol.Extensions" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-1.0.3.42" newVersion="1.0.3.42" />
```
Comment (polita) [inline TaskService/Web.config]: Same question.

File: TaskService/TaskService.csproj (hunk 3 above)
Comment (polita) [inline TaskService/TaskService.csproj]: I don't think we want <Private>True</Private> Instead, this should have <SpecificVersion>False</SpecificVersion>

File: TaskService/TaskService.csproj [hunk 6]
```diff
@@ -77,9 +77,9 @@
     <Reference Include="System.Data" />
     <Reference Include="System.Drawing" />
     <Reference Include="System.IdentityModel" />
-    <Reference Include="System.IdentityModel.Tokens.Jwt, Version=4.0.2.28, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\System.IdentityModel.Tokens.Jwt.4.0.2.202250711\lib\net45\System.IdentityModel.Tokens.Jwt.dll</HintPath>
+    <Reference Include="System.IdentityModel.Tokens.Jwt, Version=4.0.30826.1200, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.IdentityModel.Tokens.Jwt.4.0.3.308261200\lib\net45\System.IdentityModel.Tokens.Jwt.dll</HintPath>
+      <Private>True</Private>
```
Comment (polita) [inline TaskService/TaskService.csproj]: No privates.

File: TaskService/Web.config [hunk 7]
```diff
@@ -70,6 +70,10 @@
         <assemblyIdentity name="Microsoft.Owin" publicKeyToken="31bf3856ad364e35" culture="neutral" />
         <bindingRedirect oldVersion="0.0.0.0-3.0.1.0" newVersion="3.0.1.0" />
       </dependentAssembly>
+      <dependentAssembly>
```
Comment (polita) [inline TaskService/Web.config]: Why does this require a new assembly dependency?

File: TaskWebApp/TaskWebApp.csproj [hunk 8]
```diff
@@ -48,17 +48,9 @@
...
-    <Reference Include="Microsoft.Experimental.IdentityModel.Clients.ActiveDirectory.Platform, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\Microsoft.Experimental.IdentityModel.Clients.ActiveDirectory.4.0.209160138-alpha\lib\net45\Microsoft.Experimental.IdentityModel.Clients.ActiveDirectory.Platform.dll</HintPath>
-    </Reference>
-    <Reference Include="Microsoft.IdentityModel.Protocol.Extensions, Version=1.0.2.33, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\Microsoft.IdentityModel.Protocol.Extensions.1.0.2.206221351\lib\net45\Microsoft.IdentityModel.Protocol.Extensions.dll</HintPath>
+    <Reference Include="Microsoft.IdentityModel.Protocol.Extensions, Version=1.0.3.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\Microsoft.IdentityModel.Protocol.Extensions.1.0.3.308261200\lib\net45\Microsoft.IdentityModel.Protocol.Extensions.dll</HintPath>
+      <Private>True</Private>
```
Comment (polita) [inline TaskWebApp/TaskWebApp.csproj]: No privates.

File: TaskWebApp/TaskWebApp.csproj [hunk 9]
```diff
@@ -92,9 +84,9 @@
     <Reference Include="System.Data.Entity" />
     <Reference Include="System.Drawing" />
     <Reference Include="System.IdentityModel" />
-    <Reference Include="System.IdentityModel.Tokens.Jwt, Version=4.0.20622.1351, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\packages\System.IdentityModel.Tokens.Jwt.4.0.2.206221351\lib\net45\System.IdentityModel.Tokens.Jwt.dll</HintPath>
+    <Reference Include="System.IdentityModel.Tokens.Jwt, Version=4.0.30826.1200, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\packages\System.IdentityModel.Tokens.Jwt.4.0.3.308261200\lib\net45\System.IdentityModel.Tokens.Jwt.dll</HintPath>
+      <Private>True</Private>
```
Comment (polita) [inline TaskWebApp/TaskWebApp.csproj]: Same as above.

File: TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs [hunk 10]
```diff
@@ -17,16 +13,11 @@ namespace TaskService.App_Start
...
-        private readonly string _metadataEndpoint;
-
-        private readonly ReaderWriterLockSlim _synclock = new ReaderWriterLockSlim();
 
         public OpenIdConnectCachingSecurityTokenProvider(string metadataEndpoint)
         {
-            _metadataEndpoint = metadataEndpoint;
-            _configManager = new ConfigurationManager<OpenIdConnectConfiguration>(metadataEndpoint);
+            HttpClient httpClient = new HttpClient();
+            _configManager = new ConfigurationManager<OpenIdConnectConfiguration>(metadataEndpoint, httpClient);
```
Comment (brentschmaltz) [inline TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs]: nit: you could pass new HttpClient() as a parameter and not have the local variable.

**Comment by polita** [comment]
@ChenYan98 Let's finish up this PR and get it submitted so it doesn't sit for too long.

File: TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs (hunk 1 above)
Comment (ChenYan98) [reply TaskService/App_Start/OpenIdConnectCachingSecurityTokenProvider.cs]: Updated

File: TaskService/TaskService.csproj (hunk 3 above)
Comment (ChenYan98) [reply TaskService/TaskService.csproj]: Remove <Private>True</Private> and set <SpecificVersion> to false

File: TaskService/TaskService.csproj (hunk 6 above)
Comment (ChenYan98) [reply TaskService/TaskService.csproj]: Same as above

File: TaskWebApp/TaskWebApp.csproj (hunk 8 above)
Comment (ChenYan98) [reply TaskWebApp/TaskWebApp.csproj]: Same as above

File: TaskWebApp/TaskWebApp.csproj (hunk 9 above)
Comment (ChenYan98) [reply TaskWebApp/TaskWebApp.csproj]: Same as above.

**Comment by polita** [comment]
It looks like all the code changes (with the exception of reordering the using statements) was reverted. Why are all the code changes removed?

**Comment by gsacavdm** [comment]
@ChenYan98 , thank you for your contributions.
I'm closing this PR since this sample has been deprecated in place of the newer [**Azure AD B2C: Call a .NET web API from a .NET web app** sample](https://github.com/Azure-Samples/active-directory-b2c-dotnet-webapp-and-webapi). Feel free to submit PRs against that one.

**Pull request closed without merging**